qemu/tests/x86_cpu_model.py

```diff
@@ -55,6 +46,63 @@ def run(test, params, env):
         guest_model = model_pattern % ""
     else:
         test.cancel("This host doesn't support cpu model %s" % model)
+    return cpu_model, guest_model, flags
+
+
+def check_flags_guest(params, session, flags, test):
+    """
+    Check cpu flags in guest
+    :param params: Dictionary with the test parameters
+    :param session: guest session
+    :param flags: checked flags in guest
+    :param test: QEMU test object
+    """
+    if params["os_type"] == "linux":
+        error_context.context("Check cpu flags inside guest", logging.info)
+        cmd = params["check_flag_cmd"]
+        out = session.cmd_output(cmd).split()
+        missing = [f for f in flags.split() if f not in out]
+        if missing:
+            test.fail("Flag %s not in guest" % missing)
+        no_flags = params.get("no_flags")
+        if no_flags:
+            err_flags = [f for f in no_flags.split() if f in out]
+            if err_flags:
+                test.fail("Flag %s should not be present in guest" % err_flags)
+
+
+def check_flags_host(params, flags, test):
```

Comment:
Above two functions can merge to one.

Reply:
> Above two functions can merge to one.

But they need to be used separately in different place.

Reply:
```
def check_flags(session=None, ...)
    if session:
        session.cmd(cmd)
    else:
        process.run(cmd)
    ....
```

Reply:
thanks and update.